packages/@o3r/design/schematics/extract-token/index.spec.ts

```diff
@@ -66,6 +66,67 @@ $breadcrumb-pres-item-other-color: o3r.variable('breadcrumb-pres-item-other-colo
     expect(tree.readText('src/component/my-comp.theme.scss')).toBe(initialSassFile);
   });
 
+  it('should extract the Design Token with right flatten level', async () => {
+    const logger = { warn: jest.fn(), debug: jest.fn() };
+    jest.mock('@o3r/styling/builders/style-extractor/helpers', () => ({
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      CssVariableExtractor: class {
+        public extractFileContent = jest.fn<CssVariable[], any>().mockReturnValue([
+          {
+            defaultValue: '3rem',
+            name: 'breadcrumb-pres-item-icon-size',
+            label: 'breadcrumb pres item icon size',
+            type: 'string'
+          },
+          {
+            defaultValue: '#fff',
+            name: 'breadcrumb-pres-item-icon-color',
```

Comment:
won't there be a problem if you don't have the same length for variables names ?
say `breadcrumb-pres-item-icon-color` and `breadcrumb-pres-item-icon-color-bg` for example

Reply:
I guess I just misunderstood how the flattenLevel was working, I thought it was flattening from right to left for some reason